src/components/global/Button.tsx

```diff
@@ -0,0 +1,115 @@
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+import THEME from '@src/constants/Colors';
+import WIDTHS from '@src/constants/Widths';
+import { Size, Variant } from '@src/types/Button.type';
+import React, { ButtonHTMLAttributes } from 'react';
+import MoonLoader from 'react-spinners/MoonLoader';
+
+export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  variant: Variant;
+  size: Size;
+  rounder: boolean;
+  isLoading?: boolean;
+}
+
+const Button = ({ children, onClick, isLoading, variant, size, rounder }: ButtonProps) => {
+  return (
+    <StyledButton onClick={onClick} variant={variant} size={size} rounder={rounder}>
+      {/* To be fixed */}
+      {isLoading ? <MoonLoader color={THEME.PURPLE300} size={25} /> : children}
+    </StyledButton>
+  );
+};
+
+const ButtonSize = (size: Size) => {
+  if (size === 'Small') {
```

Comment:
`type alias`를 사용하면 이러한 부분에서 휴먼 에러가 나기 쉬울 것 같아요. 만약 `type alias`가 아니라 `const assertion`이었다면 아래와 같이 사용할 수 있을 것 같아요.

```suggestion
if (size === SIZE.SMALL) {
```